src/frontend/src/lib/components/swap/SwapProviderListItem.svelte

```diff
@@ -0,0 +1,78 @@
+<script lang="ts">
+	import { isNullish, nonNullish } from '@dfinity/utils';
+	import { UrlSchema } from '@dfinity/zod-schemas';
+	import type { IcTokenToggleable } from '$icp/types/ic-token-toggleable';
+	import SwapBestRateBadge from '$lib/components/swap/SwapBestRateBadge.svelte';
+	import Amount from '$lib/components/ui/Amount.svelte';
+	import Logo from '$lib/components/ui/Logo.svelte';
+	import LogoButton from '$lib/components/ui/LogoButton.svelte';
+	import { i18n } from '$lib/stores/i18n.store';
+	import type { LogoSize } from '$lib/types/components';
+	import type { OisyDappDescription } from '$lib/types/dapp-description';
+	import type { OptionAmount } from '$lib/types/send';
+	import { replacePlaceholders } from '$lib/utils/i18n.utils';
+	import { safeParse } from '$lib/validation/utils.validation';
+
+	interface Props {
+		amount: bigint;
+		token: IcTokenToggleable;
+		logoSize?: LogoSize;
+		usdBalance: OptionAmount;
+		dapp?: OisyDappDescription;
+		isBest: boolean;
+	}
+
+	const { amount, token, logoSize = 'md', usdBalance, dapp, isBest }: Props = $props();
+
+	let displayURL: string | null = $state(null);
+
+	$effect(() => {
+		if (isNullish(dapp)) {
+			displayURL = null;
+			return;
+		}
+		try {
+			const validated = safeParse({
+				schema: UrlSchema,
+				value: dapp.website
+			});
+			if (nonNullish(validated)) {
+				const url = new URL(validated);
+				displayURL = url.hostname.startsWith('www.') ? url.hostname.slice(4) : url.hostname;
+			}
+		} catch {
+			displayURL = null;
+		}
```

Comment:
instead of try-catch, you can use directly the safe parse, right? since you don't want errors raised

```suggestion
	const parsed = UrlSchema.safeParse(dapp.website);
	
	if (parsed.success) {
	        const url = new URL(parsed.data);
		displayURL = url.hostname.startsWith('www.') ? url.hostname.slice(4) : url.hostname;
		return
	}
        
        displayURL = null
```

Reply:
Nica catch. Thank you